core/opl/shovel.py

```diff
@@ -83,18 +120,178 @@ def args(parser, group_actions):
             title="opensearch",
             description="Options needed to work with OpenSearch",
         )
+        group.add_argument(
+            "--es-index",
+            default="rhtap-ci-status-data",
+            help="Elastic search index where the data will be stored",
+        )
+        group.add_argument("--data-file", help="json file to upload to elastic search")
+        group.add_argument("--end-timestamp", help="timestamp when the test ended")
 
 
 class pluginHorreum:
     def __init__(self, args):
         print("Hello from pluginHorreum init")
         self.logger = logging.getLogger("opl.showel.pluginHorreum")
 
-    def upload(self):
-        print("Hello from pluginHorreum upload")
+    def upload(self, args):
+        if args.horreum_data_file is None:
```

Comment:
Just add `required=True` to `add_argumet` calls for these options and you do not need to do this.